client/state/data-layer/wpcom/gravatar-upload/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Internal dependencies
+ */
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import {
+	bumpStat,
+	composeAnalytics,
+	recordTracksEvent,
+	withAnalytics,
+} from 'state/analytics/actions';
+
+function uploadGravatar( { dispatch }, action, next ) {
+	const { email, file } = action;
+	dispatch( http( {
+		method: 'POST',
+		path: '/gravatar-upload',
+		body: {},
+		apiNamespace: 'wpcom/v2',
+		formData: [
+			[ 'account', email ],
+			[ 'filedata', file ],
+		],
+	}, action ) );
+
+	next( action );
```

Comment:
Ah, good to know! I thought that if an action was "caught" by middleware then it was removed from the dispatch chain and would not make it to the reducers unless explicitly dispatched again.